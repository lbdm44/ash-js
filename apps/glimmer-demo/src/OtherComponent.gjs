import Component from "@glimmerx/component";

export default class OtherComponent extends Component {
  <template>
    <b>Counter Val: {{@count}}</b>
  </template>
}