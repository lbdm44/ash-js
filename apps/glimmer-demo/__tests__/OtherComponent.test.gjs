import OtherComponent from "../src/OtherComponent.gjs";
import { render, didRender } from "./util";

import { hbs } from "@glimmerx/component";
import { it, expect } from "vitest";

it("should work", async () => {
  const element = await render(OtherComponent);

  expect(element.innerHTML).toMatchSnapshot();
  expect(didRender()).toBeTruthy();
});

it("should work with args", async () => {
  const element = await render(OtherComponent, {
    args: {
      count: 10,
    },
  });

  expect(element.innerHTML).toMatchSnapshot();
  expect(didRender()).toBeTruthy();
});

it("should compile template tags", async () => {
  const element = await render(
    <template>
      <OtherComponent @count=10 />
    </template>
  );
  console.log(element.innerHTML);
  expect(element.innerHTML).toMatchSnapshot();
  expect(didRender()).toBeTruthy();
});

it("should compile hbs backtick", async () => {
  const element = await render(hbs`
    <OtherComponent @count=10 />
  `);
  console.log(element.innerHTML);
  expect(element.innerHTML).toMatchSnapshot();
  expect(didRender()).toBeTruthy();
});
