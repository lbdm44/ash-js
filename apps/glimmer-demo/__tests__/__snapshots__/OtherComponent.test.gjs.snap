// Vitest Snapshot v1

exports[`should compile hbs backtick 1`] = `
"
    
    <b>Counter Val: 10</b>
  
  "
`;

exports[`should compile template tags 1`] = `
"
      
    <b>Counter Val: 10</b>
  
    "
`;

exports[`should work 1`] = `
"
    <b>Counter Val: </b>
  "
`;

exports[`should work with args 1`] = `
"
    <b>Counter Val: 10</b>
  "
`;
